File: bitmex/hash.go
package main

import (
	"crypto/hmac"
	"crypto/sha256"
	"encoding/hex"
	"encoding/json"
	"fmt"
)

type Ticket struct {
	Symbol   string `json:"symbol"`
	OrderQty int    `json:"orderQty"`
	OrdType  string `json:"ordType"`
}

func main() {
	hs := calculateHash()
	fmt.Println(hs)
}

func calculateHash() string {

	apiSecret := "xxxxxxxxxxxxxxxxxxxxxxxx"
	verb := "POST"
	path := "/api/v1/order"
	ticket := Ticket{Symbol: "XBTUSD", OrderQty: 1, OrdType: "Market"}
	data, _ := json.Marshal(ticket)

	secret := []byte(apiSecret)
	message := []byte(verb + path + string(data))

	hash := hmac.New(sha256.New, secret)
	hash.Write([]byte(message))

	return hex.EncodeToString(hash.Sum(nil))
}

File: hash.js
var request = require('request');
var crypto = require('crypto');


var apiSecret = "BUv-Qf1yz3OTZJhBJNSeadj5AjLBqhl9jt6wROPkmKWB4QDk";

var verb = 'POST',
  path = '/api/v1/order',
  expires = new Date().getTime() + (60 * 1000), // 1 min in the future
  //data = {symbol:"XBTUSD",orderQty:1,price:3700,ordType:"Limit"};
  data = {symbol:"XBTUSD",orderQty:1,ordType:"Market"};
// Pre-compute the postBody so we can be sure that we're using *exactly* the same body in the request
// and in the signature. If you don't do this, you might get differently-sorted keys and blow the signature.
var postBody = JSON.stringify(data);
console.log(verb + path + postBody);

var signature = crypto.createHmac('sha256', apiSecret).update(verb + path + postBody).digest('hex');

console.log(signature);